### PR TITLE
(CDAP-14376) Use regional endpoint for Dataproc

### PR DESCRIPTION
**Pull request by chtyim**
- Remove the global endpoint support
- Code cleanup

### CODE REVIEW

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT7035

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java [hunk 1]
```diff
@@ -221,13 +222,13 @@ public GoogleCredential getComputeCredential() throws IOException {
...
    */
-  public GoogleCredentials getDataprocCredentials() throws IOException {
+  GoogleCredentials getDataprocCredentials() throws IOException {
     if (accountKey == null) {
       return getComputeEngineCredentials();
     }
 
     try (InputStream is = new ByteArrayInputStream(accountKey.getBytes(StandardCharsets.UTF_8))) {
-      return GoogleCredentials.fromStream(is);
+      return GoogleCredentials.fromStream(is).createScoped("https://www.googleapis.com/auth/cloud-platform");
```
Comment (anwar6953) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java]: This string is used in multiple places now; worth a constant?

Comment (chtyim) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java]: Done

**Review by anwar6953 — APPROVED** [review]
LGTM.
@albertshau care to take another look?

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java [hunk 2]
```diff
@@ -251,11 +247,25 @@ private static Network getNetworkInfo(String project, String network, Compute co
...
+    if (idx <= 0) {
+      throw new IllegalArgumentException("Invalid zone uri " + zoneUri);
+    }
+    return zoneUri.substring(idx + 1);
+  }
+
   private static ClusterControllerClient getClusterControllerClient(DataprocConf conf) throws IOException {
     CredentialsProvider credentialsProvider = FixedCredentialsProvider.create(conf.getDataprocCredentials());
 
+    String regionalEndpoint = conf.getRegion() + "-dataproc.googleapis.com:443";
```
Comment (albertshau) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java:264]: does the global endpoint also fit this pattern?

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java [hunk 3]
```diff
@@ -271,10 +275,22 @@ public static DataprocConf create(Map<String, String> properties, @Nullable SSHP
       projectId = getSystemProjectId();
     }
 
+    String region = getString(properties, "region");
+    if (region == null || AUTO_DETECT.equals(region)) {
+      region = getRegionFromZone(getSystemZone());
```
Comment (albertshau) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java]: if the zone is given by the user, it seems like this should derive the region from the provided zone and not from the system zone.

**Review by albertshau — COMMENTED** [review]
one comment

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java (hunk 2 above)
Comment (chtyim) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java:264]: No. The global endpoint is intentionally removed from the list.

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java (hunk 3 above)
Comment (chtyim) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java]: Make sense. Fixed.

**Review by albertshau — APPROVED** [review]

**Comment by chtyim** [comment]
Latest build passed

**Pull request merged**